grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -852,6 +833,29 @@ public void commit(BiConsumer<Set<Pair<String, ConceptId>>, Set<Pair<String,Conc
         }
     }
 
+    private Optional<String> commitWithLogs() throws GraknValidationException {
+        validateGraph();
+
+        boolean submissionNeeded = !getConceptLog().getInstanceCount().isEmpty() ||
+                !getConceptLog().getModifiedCastings().isEmpty() ||
+                !getConceptLog().getModifiedResources().isEmpty();
+        JSONObject conceptLog = getConceptLog().getFormattedLog();
+
+        LOG.trace("Graph is valid. Committing graph . . . ");
+        commitTransactionInternal();
+
+        //TODO: Kill when analytics no longer needs this
+        GraknSparkComputer.refresh();
+
+        LOG.trace("Graph committed.");
+
+        //No post processing should ever be done for the system keyspace
+        if(!keyspace.equalsIgnoreCase(SystemKeyspace.SYSTEM_GRAPH_NAME) && submissionNeeded) {
+            return Optional.of(conceptLog.toString());
```

Comment:
We don't return the logs if it has to do with the system graph. That's what that check is for. When the system graph is committed an empty optional is returned.